crates/sui-core/src/execution_engine.rs

```diff
@@ -511,12 +511,9 @@ fn pay<S>(
         tx_ctx,
     );
 
-    #[cfg(debug_assertions)]
-    {
-        // double check that we didn't create or destroy money
-        let new_total_coins = coins.iter().fold(0, |acc, c| acc + c.value());
-        assert_eq!(total_coins - new_total_coins, total_amount)
-    }
+    // double check that we didn't create or destroy money
+    let new_total_coins = coins.iter().fold(0, |acc, c| acc + c.value());
```

Comment:
nit: actually can you move this calculation inside `debug_assert_eq` macro? Otherwise it will always be evaluated